### PR TITLE
Proof of concept HTTPS intercept

### PR DESCRIPTION
**Pull request by claytondaley**
Converting branch to PR to make discussion easier.

### CODE REVIEW

File: app/proxy/SWProxy.js [hunk 1]
```diff
@@ -17,114 +21,27 @@ class SWProxy extends EventEmitter {
...
 
-    this.httpServer.on('connect', (req, socket) => {
-      const serverUrl = url.parse(`https://${req.url}`);
-
-      const srvSocket = net.connect(serverUrl.port, serverUrl.hostname, () => {
-        socket.write('HTTP/1.1 200 Connection Established\r\n' + 'Proxy-agent: Node-Proxy\r\n' + '\r\n');
-        srvSocket.pipe(socket);
-        socket.pipe(srvSocket);
-      });
+    proxy.listen({ port})
```
Comment (claytondaley) [inline app/proxy/SWProxy.js]: did you try `{port: port}`?  I'm not a NodeJS guy so I'm less confident about optional argument syntax.

Comment (claytondaley) [reply app/proxy/SWProxy.js]: It also accepts a `host: <host>` that you 
might want to hardcode to `0.0.0.0` while testing.

**Comment by claytondaley** [comment]
You shouldn't need to trust a CA certificate yet.  The sample code should just proxy requests (unless you go to the google address in the example.

**Comment by claytondaley** [comment]
For testing, here's a chain of potential resources:

 - [Reddit thread discussing our general need](https://www.reddit.com/r/AskNetsec/comments/9hmeam/sniffing_ssl_traffic_from_android_phones_or/) which references...
 - [Blog post about circumventing user store requirement](https://www.reddit.com/r/AskNetsec/comments/9hmeam/sniffing_ssl_traffic_from_android_phones_or/)
 - [Magisk Github project](https://github.com/NVISO-BE/MagiskTrustUserCerts) and [APKs/docs](https://forum.xda-developers.com/apps/magisk/official-magisk-v7-universal-systemless-t3473445)
 -

File: app/proxy/SWProxy.js (hunk 1 above)
Comment (Xzandro) [reply app/proxy/SWProxy.js]: `{ port }`just means `{ port: port }` in JS.
Although I doubt that the hostname is necessary, I tried it. Didn't change anything.

**Comment by Xzandro** [comment]
The proxy already does proxy everything on the specific port. The only difference on the google search page is, that this will change all links to Pwned! to demonstrate that the https interception and modyfing of these requests works fine, and so should the Summoners War ones. But there is the code 22 and the SSL errors I stated in the issue, which are the main issues of this.

**Comment by claytondaley** [comment]
OK.  I have to work but I'll try to run it locally later.

**Comment by claytondaley** [comment]
... and it works fine if you don't use the proxy i.e. the issue isn't the Nox environment?

**Comment by Xzandro** [comment]
It works fine for everything except SW, which I could see. At least in the browser I could reach all sites.

**Comment by Galunid** [comment]
I tried to investigate a bit, started capturing the traffic with wireshark. What I've found is if the used protocol is TLSv1.2, certificate exchange fails with "Certificate Unknown" error. On the other hand if TLSv1.3 is used, everything works fine. This could be a reason why only SW fails, while browsers work.

**Pull request merged**